flow/connectors/clickhouse/normalize.go

```diff
@@ -67,65 +64,140 @@ func (c *ClickhouseConnector) SetupNormalizedTable(
 	return false, nil
 }
 
+func getColName(overrides map[string]string, name string) string {
+	if newName, ok := overrides[name]; ok {
+		return newName
+	}
+	return name
+}
+
 func generateCreateTableSQLForNormalizedTable(
-	normalizedTable string,
-	tableSchema *protos.TableSchema,
-	_ string, // softDeleteColName
-	syncedAtColName string,
+	config *protos.SetupNormalizedTableBatchInput,
+	tableIdentifier string,
 ) (string, error) {
+	tableSchema := config.TableNameSchemaMapping[tableIdentifier]
+
+	var tableMapping *protos.TableMapping
+	for _, tm := range config.TableMappings {
+		if tm.DestinationTableIdentifier == tableIdentifier {
+			tableMapping = tm
+			break
+		}
+	}
+
 	var stmtBuilder strings.Builder
-	stmtBuilder.WriteString(fmt.Sprintf("CREATE TABLE IF NOT EXISTS `%s` (", normalizedTable))
+	stmtBuilder.WriteString(fmt.Sprintf("CREATE TABLE IF NOT EXISTS `%s` (", tableIdentifier))
 
+	colNameMap := make(map[string]string)
 	for _, column := range tableSchema.Columns {
 		colName := column.Name
+		dstColName := colName
 		colType := qvalue.QValueKind(column.Type)
-		clickhouseType, err := colType.ToDWHColumnType(protos.DBType_CLICKHOUSE)
-		if err != nil {
-			return "", fmt.Errorf("error while converting column type to clickhouse type: %w", err)
+		var clickhouseType string
+		var columnSetting *protos.ColumnSetting
+		if tableMapping != nil {
+			for _, col := range tableMapping.Columns {
+				if col.SourceName == colName {
+					columnSetting = col
+					if columnSetting.DestinationName != "" {
+						dstColName = columnSetting.DestinationName
+						colNameMap[colName] = dstColName
+					}
+					if columnSetting.DestinationType != "" {
+						// TODO can we restrict this to avoid injection?
```

Comment:
`slices.Contains(slices.Collect(maps.Values(qvalue.QValueKindToClickhouseTypeMap)), strings.TrimSpace(columnSetting.DestinationType))`

could further restrict to non-array types

Reply:
That will miss out on clickhouse types we don't map to like Int256